src/index.ts

```diff
@@ -352,7 +384,24 @@ class ThreadPool {
         } else {
           resolve(result);
         }
+      },
+      abortSignal);
+
+    if (abortSignal !== null) {
+      onabort(abortSignal, () => {
+        reject(new AbortError());
```

Comment:
please add a comment that we do not return because we need to do more work about the rejection.

Reply:
I’ve added a comment but I’m not sure if that’s what you hand in mind (what does “do more work about the rejection” mean?)